src/Capability/Writer.hs

```diff
@@ -69,7 +68,7 @@ import GHC.Exts (Proxy#, proxy#)
 -- prop> listen @t (m >>= k) = listen @t m >>= \(a, w1) -> listen @t (k a) >>= \(b, w2) -> pure (b, w1 `mappend` w2)
 -- prop> pass @t (tell @t w >> pure (a, f)) = tell @t (f w) >> pure a
 -- prop> writer @t (a, w) = tell @t w >> pure a
-class (Monoid w, Monad m)
+class (Monoid w, Monad m, HasStream tag w m)
```

Comment:
I think `HasStream tag w m => HasWriter tag w m` deserves a good bit of additional documentation. To me this is far from obvious. (See overall review comment).